### PR TITLE
Chore/update loading logic for layouts

### PR DESCRIPTION
**Pull request by joshenlim**
## What kind of change does this PR introduce?

- For table editor, SQL editor, database page and auth page, if the store has already been loaded, do not hide the entire UI with the loader, but let it fire "store.load" in the background
- Added some guards to prevent some errors

### CODE REVIEW

File: studio/components/layouts/AuthLayout/AuthLayout.tsx [hunk 1]
```diff
@@ -15,16 +15,15 @@ interface Props {
...
   const { ui, meta } = useStore()
-  const { isLoading, error } = meta.tables
+  const { isInitialized, isLoading, error } = meta.tables
   const projectRef = ui.selectedProject?.ref ?? 'default'
 
   const router = useRouter()
   const page = router.pathname.split('/')[4]
 
-  const [loaded, setLoaded] = useState<boolean>(false)
+  const [loaded, setLoaded] = useState<boolean>(isInitialized)
```
Comment (Ziinc) [inline studio/components/layouts/AuthLayout/AuthLayout.tsx:24]: this shouldn't be a `useState`, leaves room for possibility for local state going out of sync with store.

`loaded` should be a computed value based on `isLoading` and `isInitialized`, no?

Comment (Ziinc) [reply studio/components/layouts/AuthLayout/AuthLayout.tsx:24]: also, if say we're watching the mobx values to trigger re-renders, a more appropriate function would be `useMemo`. 

If the mobx value changes will trigger re-renders (not too sure abt this), can just do a render computation.

**Review by Ziinc — APPROVED** [review]
logic is fine, but local state syncing looks a bit error prone imo

File: studio/components/layouts/AuthLayout/AuthLayout.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/layouts/AuthLayout/AuthLayout.tsx:24]: good observation actually yeah agreed - plus this "loaded" thing is common across some pages, probably requires a small refactor on the store side

i'll do this in another PR - will leave a comment first

**Pull request merged**